api/caching/tasks.py

```diff
@@ -103,19 +103,30 @@ def ban_url(instance):
 
 
 @app.task(max_retries=5, default_retry_delay=10)
-def update_storage_usage_cache(target_id):
-    AbstractNode = apps.get_model('osf.AbstractNode')
-
-    storage_usage_total = AbstractNode.objects.get(
-        guids___id=target_id,
-    ).files.aggregate(sum=models.Sum('versions__size'))['sum'] or 0
-
-    key = cache_settings.STORAGE_USAGE_KEY.format(target_id=target_id)
+def update_storage_usage_cache(target_id, target_guid):
+    sql = """
+        SELECT sum(version.size) FROM osf_basefileversionsthrough AS obfnv
+        LEFT JOIN osf_basefilenode file ON obfnv.basefilenode_id = file.id
+        LEFT JOIN osf_fileversion version ON obfnv.fileversion_id = version.id
+        LEFT JOIN django_content_type type on file.target_content_type_id = type.id
+        WHERE file.provider = 'osfstorage'
+        AND type.model = 'abstractnode'
+        AND file.deleted_on IS NULL
+        AND file.target_object_id=%s
+    """
```

Comment:
Would it make sense to add a row limit here to minimize memory usage? Maybe something like 5000 rows, and then loop through adding the sums from each query until you've gotten all the rows. It would take a bit longer, but since it wouldn't have to re-plan the query, I doubt it would take too much longer. Plus it's a post-commit task, so it shouldn't hold up any web requests. Could be either a constant defined in the file or an optional parameter for the function with a default.